java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightStream.java

```diff
@@ -170,21 +182,17 @@ public void close() throws Exception {
    */
   public boolean next() {
     try {
-      // make sure we have the root
-      root.get().clear();
-
-      if (completed && queue.isEmpty()) {
+      if (completed.isDone() && queue.isEmpty()) {
         return false;
       }
 
-
       pending--;
       requestOutstanding();
 
       Object data = queue.take();
       if (DONE == data) {
         queue.put(DONE);
-        completed = true;
+        completed.complete(null);
```

Comment:
please document what null is here.  and if that doesn't make it clear, why is null being passed here?

Reply:
Ah, this is a `CompletableFuture<Void>`, which I clarified in a comment.